tests/D2L.CodeStyle.Analyzers.Test/Async/Generator/AsyncToSyncMethodTransformerTests.cs

```diff
@@ -65,6 +65,87 @@ internal sealed class AsyncToSyncMethodTransformerTests {
 		Assert.AreEqual( "[Blocking] void Hello() { Bar( Baz(),Quux() ); }", actual.Value.ToFullString() );
 	}
 
+	[Test]
+	public void ExplicitInterfaceSpecifier() {
+		var actual = Transform( @"[GenerateSync] async Task Foo.BazAsync() { return 6; }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Foo.Baz() { return 6; }", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void VariableDeclaration() {
+		var actual = Transform( @"[GenerateSync] async Task BarAsync() { IBaz q = await m_bazValidator.ValidateAsync( bar ); }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Bar() { IBaz q = m_bazValidator.Validate( bar ); }", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void PredefinedType() {
+		var actual = Transform( @"[GenerateSync] async Task BarAsync() { string.IsNullOrEmpty(""""); }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Bar() { string.IsNullOrEmpty(\"\"); }", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void Cast() {
+		var actual = Transform( @"[GenerateSync] async Task BarAsync() { var baz = await ( Task<int> )fooHandler.ReadFooAsync( ""foo"" ); }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Bar() { var baz = ( int )fooHandler.ReadFoo( \"foo\" ); }", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void ForEach() {
+		var actual = Transform( @"[GenerateSync] async Task BarAsync() { foreach( Task<int> fooAsync in foosAsync ) { bar += await fooAsync; } }" );

```

Comment:
Mmmm I see where your transformation of the identifier is coming from (my comments above)... interesting.
Maybe we should consistently transform all identifiers. It maybe has downsides (more likely to have name collisions etc.) but it would be very predictable.

Reply:
That's what I was thinking. It would be a bit confusing to remove some of the suffixes and not all of them